src/main/scala/com/github/agourlay/cornichon/dsl/MacroBasedDsl.scala

```diff
@@ -0,0 +1,41 @@
+package com.github.agourlay.cornichon.dsl
+
+import com.github.agourlay.cornichon.core._
+import com.github.agourlay.cornichon.macros.Macro
+
+import scala.language.experimental.{ macros ⇒ `scalac, please just let me do it!` }
+
+trait MacroBasedDsl {
+  def FeatureMacro(name: String)(body: ⇒ Scenario): FeatureDef = macro Macro.featureImpl
+
+  def ScenarioMacro(name: String)(body: ⇒ Step): Scenario = macro Macro.scenarioImpl
+  def ScenarioMacro(name: String, ignore: Boolean)(body: ⇒ Step): Scenario = macro Macro.scenarioWithIgnoreImpl
```

Comment:
yeah, as far as I can tell, named and default arguments are not supported yet. Once they were even announced, but didn't make into the scala release:

http://stackoverflow.com/a/34145464/576766